### PR TITLE
Add PEM_write_bio_DHparams binding

### PR DESCRIPTION
**Pull request by evilaliv3**
This pull request simply add PEM_write_bio_DHparams binding.

Adding this binding would be really valuable in order to make possible to generate DH parameters from the application and save them on a the filesystem using the PEM format.

### CODE REVIEW

**Comment by alex** [comment]
Jenkins, ok to test.

**Comment by evilaliv3** [comment]
what do you mean @alex? it's a not relevant error right?

**Comment by alex** [comment]
@evilaliv3 that command was for Jenkins, to test it to run the tests. I haven't looked at why they're failing.

**Comment by reaperhulk** [comment]
This is a consequence of the signature of this macro changing over time. In the most recent versions of OpenSSL it is declared with a const second argument, while older versions did not do this. To fix this move the declaration to the MACROS section (where we don't verify header declarations).

**Comment by evilaliv3** [comment]
i see @reaperhulk, thanks.

going to do it and rebase the patch for a matter of clearness.

File: src/_cffi_src/openssl/pem.py [hunk 1]
```diff
@@ -72,13 +72,13 @@
 int PEM_write_bio_RSAPublicKey(BIO *, const RSA *);
 
 EVP_PKEY *PEM_read_bio_PUBKEY(BIO *, EVP_PKEY **, pem_password_cb *, void *);
-int PEM_write_bio_PUBKEY(BIO *, EVP_PKEY *);
```
Comment (alex) [inline src/_cffi_src/openssl/pem.py]: Why was this removed?

**Comment by alex** [comment]
Not sure what happened here, several extra commits appear to have been added in the rebase process.

File: src/_cffi_src/openssl/pem.py (hunk 1 above)
Comment (evilaliv3) [inline src/_cffi_src/openssl/pem.py]: just error :) you are too fast!

i'm also rebasing against the current master

**Comment by evilaliv3** [comment]
here we are @alex! the patch should be now correct; let's wait for the tests to finish!
